driver/bpf/plumbing_helpers.h

```diff
@@ -92,31 +92,46 @@ static __always_inline long bpf_syscall_get_retval(void *ctx)
 static __always_inline long bpf_syscall_get_nr(void *ctx)
 {
 	struct sys_enter_args *args = (struct sys_enter_args *)ctx;
-	long id;
+	long id = 0;
 
 #ifdef BPF_SUPPORTS_RAW_TRACEPOINTS
 
 	struct pt_regs *regs = (struct pt_regs *)args->regs;
 
-#ifdef CONFIG_ARM64
-	
+#ifdef CONFIG_X86_64
+
+	/* See here for the definition:
+	 * https://github.com/torvalds/linux/blob/69cb6c6556ad89620547318439d6be8bb1629a5a/arch/x86/include/asm/syscall.h#L40
+	 */	
+	id = _READ(regs->orig_ax);
+
+#elif CONFIG_ARM64
+
+	/* See here for the definition:
+	 * https://github.com/torvalds/linux/blob/69cb6c6556ad89620547318439d6be8bb1629a5a/arch/arm64/include/asm/syscall.h#L23
+	 */	
 	id = _READ(regs->syscallno);
 
+#elif CONFIG_S390
+
+	/* See here for the definition:
+	 * https://github.com/torvalds/linux/blob/69cb6c6556ad89620547318439d6be8bb1629a5a/arch/s390/include/asm/syscall.h#L24
+	 */
+	id = _READ(regs->int_code);
+	id = id & 0xffff;
+
 #else 
 	
-	/* Right now we used this for all other architectures that are not ARM
-	 * becuase we support only `x86`, but we have to check if it is correct
-	 * in case of other architectures.
-	 */
-	id = _READ(regs->orig_ax);
+	/* Unknown architecture. */
+	id = 0;
```

Comment:
[ultra-nit] This assignment is redundant, `id` is already initialized in line 95.

Reply:
yeah I left them to make it more explicit that we have an `unknown architecture case` but I understand your point, probably is better to have a clearer code, I will remove that, thank you for it as always :)

Reply:
Solved in the last commit :)